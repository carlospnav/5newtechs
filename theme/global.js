import { injectGlobal } from "styled-components"

injectGlobal`
  html {
    overflow-x: hidden;
    overflow-y: auto;
    width: 100%;
  }

  body {
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    margin: 0;
    overflow-y: auto;
    width: 100%;
    background-color: black;
  }

  * {
    min-height: 0;
    min-width: 0;
  }

  *, *:before, *:after {
    box-sizing: border-box;
  }

  a {
    text-decoration: none;
  }

`
