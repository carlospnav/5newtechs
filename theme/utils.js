import { css } from "styled-components"

export const media = {
  desktop: (...args) => css`
    @media (max-width: 1280px) {
      ${css(...args)}
    }
  `
}

export function flexGroup(
  direction = "row",
  justifyContent = "flex-start",
  alignItems = "flex-start",
  display = "flex"
) {
  return `
    flex-direction: ${direction};
    justify-content: ${justifyContent};
    align-items: ${alignItems};
    display: ${display};
  `
}

export function romanizeEpisodeNum(num) {
  const lookup = { X:10, IX:9, V:5, IV:4, I:1 };
  let roman = '';

  if (isNaN(Number(num)))
    return "Num could not be romanized. Are you sure num is a number?"

  for ( let i in lookup ) {
    while ( num >= lookup[i] ) {
      roman += i;
      num -= lookup[i];
    }
  }
  return roman;
}

export function createFullEpisodeTitle(episode, title){
  if (episode !== undefined && isNaN(Number(episode)) || !title){
    return "Oops. Could not create the title with the information provided. It's a trap!"
  }
  return `Star Wars:${episode ? ` Episode ${romanizeEpisodeNum(episode)}` : ""}: ${title}`
}