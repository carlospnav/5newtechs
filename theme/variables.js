export default {
  color: {
    borderColor: "#343434",
    sectionTitle: "#4e5760",
    link: "#007da9"
  },
  font: {
    roboto: `'Roboto', sans-serif`
  },
  fontWeight: {
    regular: 400,
    semibold: 600,
    bold: 700
  },
  zindex: {}
}
