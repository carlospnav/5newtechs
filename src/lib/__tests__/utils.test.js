import { createFullEpisodeTitle, romanizeEpisodeNum, flexGroup, media } from "theme/utils"

describe("createFullEpisodeTitle", () => {

  it("should properly return the full title if provided with correct episode and title values", () => {
    expect(createFullEpisodeTitle("3", "A New Hope")).toEqual("Star Wars: Episode III: A New Hope")
  })

  it("should properly return a half title if provided with only a title", () => {
    expect(createFullEpisodeTitle(undefined, "The Last Jedi")).toEqual("Star Wars:: The Last Jedi")
  })

  it("should return an error message if episode isn't a proper number. ", () => {
    expect(createFullEpisodeTitle("Han", "The Last Jedi")).toEqual("Oops. Could not create the title with the information provided. It's a trap!")
  })
})

describe("romanizeEpisodeNum", () => {
  it("should convert the number 2 into II.", () => {
    expect(romanizeEpisodeNum(2)).toEqual("II")
  })

  it("should convert the number 4 into IV.", () => {
    expect(romanizeEpisodeNum(4)).toEqual("IV")
  })

  it("should convert the number 19 into IX.", () => {
    expect(romanizeEpisodeNum(19)).toEqual("XIX")
  })

  it("should return an error message if number provided is not a number.", () => {
    expect(romanizeEpisodeNum("Han")).toEqual("Num could not be romanized. Are you sure num is a number?")
  })
})

describe("flexGroup", () => {
  it("should return a proper row flex item template string", () => {
    expect(flexGroup(null, "space-between", "center")).toMatchSnapshot()
  })

  it("should return a proper column flex item template string.", () => {
    expect(flexGroup("column", "center")).toMatchSnapshot()
  })
})

describe("media", () => {
  it("should return a proper template string with the extra CSS rules provided", () => {
    expect(media.desktop`
      height: 200px;
      width: 100px;`
    ).toMatchSnapshot()
  })

  it("should return a proper template string with the extra CSS rules provided", () => {
    expect(media.desktop`
      height: 290px;
      width: 170px;
      background: #FFFFFF`
    ).toMatchSnapshot()
  })
})

