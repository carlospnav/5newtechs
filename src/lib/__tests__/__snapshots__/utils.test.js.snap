// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flexGroup should return a proper column flex item template string. 1`] = `
"
    flex-direction: column;
    justify-content: center;
    align-items: flex-start;
    display: flex;
  "
`;

exports[`flexGroup should return a proper row flex item template string 1`] = `
"
    flex-direction: null;
    justify-content: space-between;
    align-items: center;
    display: flex;
  "
`;

exports[`media should return a proper template string with the extra CSS rules provided 1`] = `
Array [
  "
    @media (max-width: 1280px) {
      ",
  "
      height: 200px;
      width: 100px;",
  "
    }
  ",
]
`;

exports[`media should return a proper template string with the extra CSS rules provided 2`] = `
Array [
  "
    @media (max-width: 1280px) {
      ",
  "
      height: 290px;
      width: 170px;
      background: #FFFFFF",
  "
    }
  ",
]
`;
